### PR TITLE
Error when no start command is found

### PR DESCRIPTION
**Pull request by Milo123459**
This PR errors when no start command can be found, as opposed to giving an un-descriptive error message. This should hopefully be clearer for the end user.

### CODE REVIEW

File: src/nixpacks/builder/docker/dockerfile_generation.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ impl DockerfileGenerator for StartPhase {
     ) -> Result<String> {
         let start_cmd = match &self.cmd {
             Some(cmd) => utils::get_exec_command(cmd),
-            None => "".to_string(),
+            None => bail!("No start command cound be found"),
```
Comment (coffee-cup) [inline src/nixpacks/builder/docker/dockerfile_generation.rs:235]: The dockerfile generation should not error. It should error when the build plan is created/validated

File: src/nixpacks/app.rs [hunk 2]
```diff
@@ -171,10 +171,7 @@ impl App {
 
     /// Convert an absolute path to a path relative to the app source directory
     pub fn strip_source_path(&self, abs_path: &Path) -> Result<PathBuf> {
-        let source_str = match self.source.to_str() {
-            Some(s) => s,
-            None => bail!("Failed to parse source path"),
-        };
+        let source_str = self.source.to_str().unwrap_or("");
```
Comment (coffee-cup) [inline src/nixpacks/app.rs:174]: Why default to empty string here?

Comment (Milo123459) [reply src/nixpacks/app.rs:174]: Oops, got left in.

Comment (Milo123459) [reply src/nixpacks/app.rs]: Fixed in [`c236391` (#507)](https://github.com/railwayapp/nixpacks/pull/507/commits/c2363914f6b5e09ed8ad050499fd5a29d492a59b)

File: src/nixpacks/app.rs [hunk 3]
```diff
@@ -171,10 +171,7 @@ impl App {
 
     /// Convert an absolute path to a path relative to the app source directory
     pub fn strip_source_path(&self, abs_path: &Path) -> Result<PathBuf> {
-        let source_str = match self.source.to_str() {
-            Some(s) => s,
-            None => bail!("Failed to parse source path"),
-        };
+        let source_str = self.source.to_str().unwrap();
```
Comment (coffee-cup) [inline src/nixpacks/app.rs:174]: Why change it from what it was before? Which was a `bail!` with error message

Comment (coffee-cup) [reply src/nixpacks/app.rs:174]: Unwrapping will also panic as opposed to returning a `Result`

**Review by coffee-cup — COMMENTED** [review]
What does it look like if there is no start command found? I think we might want to
1. Only error if building and there is no start command (not while running the plan command). Happy to hear opposing thoughts though
2. Show the build plan in the current state if there is no start command found

File: src/nixpacks/app.rs (hunk 3 above)
Comment (Milo123459) [reply src/nixpacks/app.rs]: Resolved

**Comment by coffee-cup** [comment]
It actually might make sense to move the start command check here https://github.com/railwayapp/nixpacks/blob/main/src/lib.rs#L95

and then also print the build plan before bailing

**Comment by Milo123459** [comment]
What is the benefit of moving it there?

**Comment by coffee-cup** [comment]
> What is the benefit of moving it there?

It will only error when actually building and won't error when running `nixpacks plan`

**Comment by coffee-cup** [comment]
It is also not really the responsibility of the build plan generator to check if a start command exists.

**Comment by Milo123459** [comment]
Done, moved logic.

**Review by coffee-cup — APPROVED** [review]

**Pull request merged**